antarest/core/tasks/service.py

```diff
@@ -94,11 +104,17 @@ def await_task(self, task_id: str, timeout_sec: int = DEFAULT_AWAIT_MAX_TIMEOUT)
 
 
 # noinspection PyUnusedLocal
-def noop_notifier(message: str) -> None:
-    """This function is used in tasks when no notification is required."""
+class NoopNotifier(ITaskNotifier):
+    """This class is used in tasks when no notification is required."""
 
+    def __call__(self, message: str) -> None:
+        return
 
-class TaskJobLogRecorder:
+    def notify_progress(self, progress: int) -> None:
+        return
+
+
+class TaskJobLogRecorder(ITaskNotifier):
```

Comment:
We can rename it to TaskLogAndProgressRecorder maybe ? ugly but that's what it does ...